operator/v1/types.go

```diff
@@ -214,3 +214,21 @@ type NodeStatus struct {
 	// lastFailedRevisionErrors is a list of the errors during the failed deployment referenced in lastFailedRevision
 	LastFailedRevisionErrors []string `json:"lastFailedRevisionErrors"`
 }
+
+// CertificateRotation contains information to control the timing of cert rotation.  You cannot specify rules here,
+// only the times.  Default values for rotation timing can vary by the particular certificate being rotated.
+// Generally these should not be touched.  The names are implementation sensitive and we expect to change the overall
+// mechanism in a future release.
+type CertificateRotation struct {
+	// signerRotationCadence controls how often the signer will rotate
+	SignerRotationCadence CertificateRotationCadence `json:"signerRotationCadence"`
+	// targetRotationCadence controls how often the target will rotate
+	TargetRotationCadence CertificateRotationCadence `json:"targetRotationCadence"`
+}
+
+type CertificateRotationCadence struct {
+	// validity is how long the certificate will be valid for
+	Validity metav1.Duration `json:"validity,omitempty"`
+	// refreshPercentage is what percentage into the validity duration, the certificate gets rotated
+	RefreshPercentage float32 `json:"refreshPercentage,omitempty"`
```

Comment:
Do we really need to support anything other than whole numbers?

Reply:
> Do we really need to support anything other than whole numbers?

hmmm... Clearly my doc is bad.  This 0-1.  Think users prefer 0-100?  heathens.

Reply:
if we say percentage, i think we should use whole numbers (0.5% vs. 50%)

Reply:
Maybe use `intstr.IntOrString` like we do in deployments when expressing surge and unavailability.

Reply:
I'd like to see an example like follows in the doc:
```suggestion
Example: when the certificate validity is set to 8h, and refreshPersentage is set to 75%, the certificates will rotate after 6h. 
```
It is obvious for us, but simple example never hurts.

Reply:
I'm still not in favor of the name, though, but I can't think of anything better... maybe just `MaxValidity`?

Reply:
> I'm still not in favor of the name, though, but I can't think of anything better... maybe just `MaxValidity`?

Instead of a percentage have a separate duration?  I'm ok with that.

Reply:
With `intstr.IntOrString` you still get percentage, it's just coded differently. As for the name `MaxValidity` is by name connected with Validity which is duration itself.